File: simple-statemachine/simple-statemachine-java/src/test/java/com/example/statemachine/simple/TestArgumentsHelper.java
package com.example.statemachine.simple;

import static java.util.function.Predicate.not;

import com.example.statemachine.simple.doument.DocumentEvent;
import java.util.Arrays;
import java.util.Set;
import java.util.stream.Stream;
import org.junit.jupiter.params.provider.Arguments;

public class TestArgumentsHelper {
  public static Stream<Arguments> getDocumentEvents(final Set<DocumentEvent> exclusions) {

    return Arrays.stream(DocumentEvent.values())
      .filter(not(exclusions::contains))
      .map(Arguments::of);
  }

}

File: simple-statemachine-with-persistence/using-statemachine-recipe/simple-statemachine-kotlin/src/test/kotlin/com/example/statemachine/simple/TestArgumentsHelper.kt
package com.example.statemachine.simple

import com.example.statemachine.simple.document.DocumentEvent


object TestArgumentsHelper {
  fun getDocumentEvents(exclusions: Set<DocumentEvent>): List<DocumentEvent> {
    return DocumentEvent.entries.toTypedArray()
      .filter { event -> !exclusions.contains(event) }
      .toList()
  }
}
